### PR TITLE
GEODE-4009: add jdbc connection pooling

### PR DESCRIPTION
**Pull request by dschneider-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: gradle/dependency-versions.properties [hunk 1]
```diff
@@ -32,6 +32,8 @@ commons-lang.version = 2.6
 commons-logging.version = 1.2
 commons-modeler.version = 2.0.1
 commons-validator.version = 1.6
+//commons-dbcp2.version = 2.1.1
```
Comment (nreich) [inline gradle/dependency-versions.properties]: Need to remove this line?

File: geode-connectors/src/test/java/org/apache/geode/connectors/jdbc/internal/DataSourceManagerUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  private JdbcDataSource dataSource;
+  private JdbcDataSource dataSource2;
+
+  private ConnectionConfiguration connectionConfig;
+  private ConnectionConfiguration connectionConfig2;
+
+  @Before
+  public void setup() throws Exception {
+    dataSource = mock(JdbcDataSource.class);
+    JdbcDataSourceFactory dataSourceFactory = new JdbcDataSourceFactory() {
```
Comment (nreich) [inline geode-connectors/src/test/java/org/apache/geode/connectors/jdbc/internal/DataSourceManagerUnitTest.java]: Should this just be a mock that returns dataSource when create is called (with the right ConnectionCongifuration)?

File: geode-connectors/src/test/java/org/apache/geode/connectors/jdbc/internal/DataSourceManagerUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  public void retrievesSameDataSourceForSameConnectionConfig() throws Exception {
+    JdbcDataSource returnedDataSource = manager.getDataSource(connectionConfig);
+    JdbcDataSource secondReturnedDataSource = manager.getDataSource(connectionConfig);
+
+    assertThat(returnedDataSource).isNotNull().isSameAs(dataSource);
+    assertThat(secondReturnedDataSource).isNotNull().isSameAs(dataSource);
+  }
+
+  private void registerTwoDataSourceFactory() {
+    JdbcDataSourceFactory dataSourceFactory = new JdbcDataSourceFactory() {
```
Comment (nreich) [inline geode-connectors/src/test/java/org/apache/geode/connectors/jdbc/internal/DataSourceManagerUnitTest.java]: Use mock that returns different value depending on ConnectionConfiguration given?

File: geode-connectors/build.gradle [hunk 4]
```diff
@@ -18,6 +18,8 @@
 dependencies {
     compile project(':geode-core')
     compile project(':geode-common')
+    //compile group: 'com.zaxxer', name: 'HikariCP', version: '2.7.4'
```
[the remark this answers is not in the document]
Comment (dschneider-pivotal) [reply geode-connectors/build.gradle]: done

File: gradle/dependency-versions.properties (hunk 1 above)
Comment (dschneider-pivotal) [reply gradle/dependency-versions.properties]: done

**Comment by dschneider-pivotal** [comment]
latest changes address all review comments

**Pull request merged**